scripts/add-checklist-to-pr.js

```diff
@@ -0,0 +1,63 @@
+const fs = require('fs');
+
+module.exports = async ({ github, context }) => {
+    const pr_number = context.issue.number;
+    const marker = '## Samples Checklist';
+    
+    let checklist;
+    let error = false;
+
+    try {
+      // Read the checklist from the file
+      checklist = fs.readFileSync('checklist.txt', 'utf8');
+      checklist = checklist.trim();
+      if(!checklist) {
+        checklist = "✅ All good!"
+      }
+      else {
+        error = true;
+      }
+    } catch (readFileError) {
+      throw new Error("Could not read samples checklist from file. Please fix the issues and try again.")
+    }
+
+    try {
+      // Get the current PR
+      const { data: pullRequest } = await github.rest.pulls.get({
+          owner: context.repo.owner,
+          repo: context.repo.repo,
+          pull_number: pr_number
+      });
+    } catch (getPrError) {
+        throw new Error ("Could not get current PR from source. Please fix the issues and try again.")
+    }
+
+    try {
+      let newBody;
+      const body = pullRequest.body || "";
+      const markerIndex = body.indexOf(marker);
+
+      if (markerIndex !== -1) {
+          // Replace the content below the marker
+          newBody = body.substring(0, markerIndex + marker.length) + "\n" + checklist;
+      } else {
+          // Append the checklist if the marker doesn't exist
+          newBody = body + "\n" + marker + "\n" + checklist;
+      }
+
+      // Update the PR description
+      await github.rest.pulls.update({
+          owner: context.repo.owner,
+          repo: context.repo.repo,
+          pull_number: pr_number,
+          body: newBody
+      });
+    } catch (updatePrError) {
+      throw new Error("Could not update PR description based on samples checklist. Please fix the issues and try again.")
+    }
+
+    if (error) {
+      throw new Error("Incomplete samples checklist. Please fix the issues and try again.");
+    }
+    
+}
```

Comment:
Let's add a newline to the end of the file here: 
```suggestion
}

```

You can configure vscode to add it for you automatically by adding this to your settings. Search for `final` and these two checkboxes will come up
<img width="912" alt="Screenshot 2024-12-19 at 2 42 12 PM" src="https://github.com/user-attachments/assets/60e1d199-adef-4008-9bb3-96cfeab47e42" />

If you're curious about why a file should end with an empty line, see https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
Github shows you an icon at the end of the file if its missing, that's how I knew! (see https://stackoverflow.com/questions/65300309/add-missing-trailing-newline-to-all-files-in-git-repo)